src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -183,8 +232,20 @@ impl DorisInsertClient {
             .map_err(|err| SinkError::Http(err.into()))?;
 
             if status == StatusCode::OK && !raw_string.is_empty() {
-                let response: DorisInsertResultResponse =
-                    serde_json::from_str(&raw_string).map_err(|err| SinkError::Http(err.into()))?;
+                let response = if send_type.eq(DORIS) {
+                    let response: DorisInsertResultResponse = serde_json::from_str(&raw_string)
+                        .map_err(|err| SinkError::Http(err.into()))?;
+                    InsertResultResponse::Doris(response)
+                } else if send_type.eq(STARROCKS) {
+                    let response: StarrocksInsertResultResponse = serde_json::from_str(&raw_string)
+                        .map_err(|err| SinkError::Http(err.into()))?;
+                    InsertResultResponse::Starrocks(response)
+                } else {
+                    return Err(SinkError::Http(anyhow::anyhow!(
+                        "Can't convert {:?}'s http response to struct",
+                        send_type
+                    )));
+                };
```

Comment:
Because the only difference between sr and doris is how to parse the response, I think it is cleaner to keep the original implementation of `InserterBuilder` without letting it understand the `send_type`. `insert.finish` can return `Result<Bytes>`. `DorisInserter` and `StarrocksInserter` can parse the Bytes response accordingly.